test/testcommon/src/main/java/io/pravega/test/common/TestUtils.java

```diff
@@ -58,7 +58,6 @@ public synchronized static int getAvailableListenPort() {
                 return candidatePort;
             } catch (IOException e) {
                 // Do nothing. Try another port.
-                NEXT_PORT.addAndGet(RAND.nextInt());
```

Comment:
This line is intentional because it prevents the case where two rngs from competing tests running in parallel endup on the same sequence, this jumps them apart.

Reply:
NEXT_PORT is initialized to random value and from that point onwards we always allocate ports sequentially/monotonically increasing numbers.

The port numbers will wrap around after every 28K calls. This means it is unlikely that we'll end up using already used port. Even if there is laggard test that is holding on a port or two, the lop will simply skip over sequentially.

Reply:
yes but remember we launch 10 tests in parallel. So there are 10 random points and 28k ports to choose from. Every hundred runs or so two of the starting points are going to be near each other by chance and then a lower one could overtake a higher one. The code here is to reset the situation if that happens, which should prevent real errors if this has happened. If the collision is spurious with some other random port it won't matter either way.

Reply:
Is it a requirement that all ports allocated for a test should be strictly consecutive? 
If not, then it probably should not matter whether there are parallel tests because here we'll always increment an AtomicInt which guarantees each test will get unique set of port numbers even if they are interleaved.